libp2p-identify/src/protocol.rs

```diff
@@ -133,13 +145,30 @@ where
 					.map(|(msg, _)| msg)
 					.map_err(|(err, _)| err)
 					.and_then(|msg| {
+						debug!(target: "libp2p-identify", "Received identify message from {:?}",
+							   observed_addr_log.expect("debug log level is enabled"));
```

Comment:
This error message could be better. My choice would be ``Programmer error: expected `observed_addr_log' to be non-null since debug log level is enabled`` but if you have a better one then you should use that.